Fix stdlib deprecation warnings

Hexta opened the pull request:
Use validate_legacy function instead of deprecated validate_* functions.
Maybe it is better to use Puppet rich data types instead of validate_* functions.

Zenidd commented:
Hey @Hexta, how you doing? 

I'm reviewing it soon. Thank you for your new contribution, you are making wazuh-puppet a better repo.

_-  JP_

Zenidd reviewed (commented):
Hey @Hexta, 

What is the motivation behind adding the `.fixtures.yml` file? Is it only useful in RSpec tests environments?

Hexta commented:
> Hey @Hexta,
> 
> What is the motivation behind adding the `.fixtures.yml` file? Is it only useful in RSpec tests environments?

Hi @Zenidd 

I've added this file to ensure that RSpec tests will be running against stdlib v4.13.0 (with deprecated `validate_<type>()` functions and introduced `validate_legacy()`).

Zenidd commented:
Understood @Hexta, I'm appending here the [related changelog](https://forge.puppet.com/modules/puppetlabs/stdlib/4.18.0/changelog#supported-release-4130).

_- JP_
